js/client-battle.js

```diff
@@ -1332,6 +1332,30 @@
 			var basePower = move.basePower;
 			var basePowerComment = '';
 			var thereIsWeather = (this.battle.weather in {'sunnyday': 1, 'desolateland': 1, 'raindance': 1, 'primordialsea': 1, 'sandstorm': 1, 'hail':1});
+			if (myPokemon.item) {
+				var item = Tools.getItem(myPokemon.item).name;
+				var pos = this.choice.choices.length;
+				var moveType = this.getMoveType(move, this.battle.mySide.active[pos]);
+				var itemInfo = item.split(' ');
+				var name = move.name;
+				if (itemInfo[1] == 'Gem') {
+					if (moveType == itemInfo[0]) {
+						if (basePower) {
+							if (name != 'Struggle' || name != 'Water Pledge' || name != 'Grass Pledge' || name != 'Fire Pledge' || name != 'Fling' ) {
+								if (this.battle.gen == 6) {
+									basePower *= 1.3;
+								}
+								else if (this.battle.gen == 5) {
+									basePower *= 1.5;
+								}
```

Comment:
A more robust variant would be:
``basePower *= this.battle.gen >= 6 ? 1.3 : 1.5``

Thus handling the likely possibility of gems not changing its effect in the future, and supporting gems in DPP and below (Custom Game).